service/s3/s3crypto/crypto.go

```diff
@@ -41,3 +42,11 @@ func (rc *CryptoReadCloser) Read(b []byte) (int, error) {
 	}
 	return rc.Decrypter.Read(b)
 }
+
+type bytesWriteSeeker struct {
+	bytes.Buffer
+}
+
+func (ws *bytesWriteSeeker) Seek(offset int64, whence int) (int64, error) {
```

Comment:
Yea, I would still need to wrap the `aws.WriteAtBuffer`, due to it not supporting `Write` when using `io.Copy`